### ISSUE TITLE
Return self from `despawn_descendants`

### ISSUE THREAD

**Issue by Shatur** [bevyengine/bevy#8883]
## What problem does this solve or what need does it fill?

Sometimes you want to despawn all descendants and insert new children. In order to do it you need to do something like this:
```rust
let mut entity = commands.entity(entity);
entity.despawn_descendants();
entity.with_children(|parent| {
    // Spawn new children.
});
```

## What solution would you like?

It would be great to return `self` from `despawn_descendants` in order to do something like this:
```rust
commands.entity(entity).despawn_descendants().with_children(|parent| {
    // Spawn new children.
});
```

**Comment by AnthonyKalaitzis** [comment]
I wouldn't mind giving this one a go if its alright

**Comment by Shatur** [comment]
@AnthonyKalaitzis feel free to do it!

**Comment by Shatur** [comment]
Closed by #8928.
